Google.PowerShell/Compute/GceManagedInstanceGroupCmdlets.cs

```diff
@@ -393,40 +418,82 @@ protected override void ProcessRecord()
             InstanceGroupManager manager;
             switch (ParameterSetName)
             {
-                case ParameterSetNames.ByProperies:
+                case ParameterSetNames.ByZoneProperties:
+                case ParameterSetNames.ByRegionProperties:
                     manager = new InstanceGroupManager
                     {
                         Name = Name,
                         InstanceTemplate = InstanceTemplate,
-                        Zone = Zone,
                         BaseInstanceName = BaseInstanceName ?? Name,
                         Description = Description,
                         TargetPools = TargetPool,
                         NamedPorts = BuildNamedPorts(),
                         TargetSize = TargetSize
                     };
                     break;
-                case ParameterSetNames.ByObject:
+                case ParameterSetNames.ByRegionObject:
+                case ParameterSetNames.ByZoneObject:
                     manager = Object;
                     break;
                 default:
                     throw UnknownParameterSetException;
             }
+
+            bool isRegionInstanceGroup = ParameterSetName == ParameterSetNames.ByRegionObject
```

Comment:
What happens if the input object already has the Region or Zone (or even Project) fields filled out? Do they get overwritten? It looks like a bug that already existed, but is exacerbated by these changes.

Reply:
Yes, they will get overwritten. I implemented this way because I think there's a use case where the user has an instance group in region A and he wants to make a copy of that in region B, so he would get the existing instance group and pipe it to the cmdlet and just supply `-Region B`.

Reply:
Maybe a better solution will be to get the existing Region or Zone in the object and overwrite it if the user supply something via `-Region`, `-Zone` or `-Project`.